compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -7469,13 +7469,114 @@ OMR::X86::TreeEvaluator::vmushrEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 TR::Register*
 OMR::X86::TreeEvaluator::vrolEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return TR::TreeEvaluator::vectorBinaryArithmeticEvaluator(node, cg);
+   TR::DataType type = node->getType();
+   TR::DataType et = type.getVectorElementType();
+   TR::VectorLength vl = type.getVectorLength();
+
+   TR_ASSERT_FATAL(et != TR::Int8 && !et.isFloatingPoint(), "Vector rotate evaluator does not support 8-bit types nor floating-point types");
+
+   if (cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_AVX512F) && et != TR::Int16)
+      {
+      return TR::TreeEvaluator::vectorBinaryArithmeticEvaluator(node, cg);
+      }
+   else if (cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_AVX2))
+      {
+      TR_ASSERT_FATAL(vl != TR::VectorLength512 || cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_AVX512F) , "512-bit vector rotate not supported without AVX2");
```

Comment:
The assert message doesn't make sense to me. Isn't this saying that 512-bit vector rotate is not supported without AVX-512F ?